src/adtypes_interface.jl

```diff
@@ -133,18 +134,60 @@ julia> hessian_sparsity(f, [1.0, 2.0, 3.0, 4.0], TracerLocalSparsityDetector())
 """
 struct TracerLocalSparsityDetector{TG <: GradientTracer, TH <: HessianTracer} <:
     ADTypes.AbstractSparsityDetector end
-function TracerLocalSparsityDetector(
-        ::Type{TG}, ::Type{TH}
-    ) where {TG <: GradientTracer, TH <: HessianTracer}
-    return TracerLocalSparsityDetector{TG, TH}()
+
+#==========================#
+# Convenience constructors #
+#==========================#
+
+for D in (:TracerSparsityDetector, :TracerLocalSparsityDetector)
+    # Specity both tracer types as arguments
+    @eval ($D)(::Type{TG}, ::Type{TH}) where {TG <: GradientTracer, TH <: HessianTracer} = ($D){TG, TH}()
+
+    # Only specify one tracer type, fall back to default for other
```

Comment:
Can we choose the other type in a more clever way if we know `G` or `H`?

Reply:
I'm not sure what you mean by this.